powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java

```diff
@@ -229,31 +230,37 @@ public static JsonSchema getJsonSchema(String schema) {
     public static JsonSchema getJsonSchema(String schema, boolean validateSchema) {
         JsonSchema jsonSchema = schemas.get(schema);
 
-        if (jsonSchema == null) {
-            if (schema.startsWith(CLASSPATH)) {
-                String filePath = schema.substring(CLASSPATH.length());
-                InputStream schemaStream = ValidationAspect.class.getResourceAsStream(filePath);
+        if (jsonSchema != null) {
+            return jsonSchema;
+        }
+
+        if (schema.startsWith(CLASSPATH)) {
+            String filePath = schema.substring(CLASSPATH.length());
+            try (InputStream schemaStream = ValidationAspect.class.getResourceAsStream(filePath)) {
                 if (schemaStream == null) {
                     throw new IllegalArgumentException("'" + schema + "' is invalid, verify '" + filePath + "' is in your classpath");
                 }
+
                 jsonSchema = ValidationConfig.get().getFactory().getSchema(schemaStream);
-            } else {
-                jsonSchema = ValidationConfig.get().getFactory().getSchema(schema);
+            } catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)